unity-client/Assets/Scripts/MainScripts/DCL/Controllers/CharacterController/DCLCharacterController.cs

```diff
@@ -203,6 +207,10 @@ public void SetPosition(Vector3 newPosition)
                 ReportMovement();
 
             OnCharacterMoved?.Invoke(characterPosition);
+
+            float distance = Vector3.Distance(characterPosition.worldPosition, lastPosition);
+            if (distance != 0f && isGrounded)
```

Comment:
comparing floats by equality can introduce precision issues, so I suggest to use the `>` operator.

```suggestion
            if (distance > 0f && isGrounded)
```